src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -385,7 +392,7 @@ export class VisualizeEmbeddable
       searchContext: {
         timeRange: this.timeRange,
         query: this.input.query,
-        filters: this.input.filters,
+        filters: this.mergeSearchSourceFiltersWith(this.input.filters),
```

Comment:
I haven't reviewed/tested yet, but since this changes `visualize_embeddable.ts ` it might be that the same fix is needed in other embeddable (saved search / lens / maps)  

Not sure if you have tested those?

Reply:
I would imagine that the changes would rather live inside filter manager code rather in individual embeddables.

Reply:
Not sure how filter manager helps here. 
As I understand, we have filters that are coming from different sources on an embeddable layer and then we need to merge them.

Reply:
I haven't looked too deeply into this, but to me this looks like a classic case of `mapAndFlattenFilters`

Reply:
Let me take a deeper look

Reply:
Right, I tested saved search and lens, they seemed to work correctly, I haven't tested maps though!